scripts/mingw-build.bat

```diff
@@ -0,0 +1,6 @@
+@echo off
+pushd ..\microx
+g++ -g -shared Executor.cpp Python.cpp -Lc:\Python37-64\libs -lpython37 -std=c++14 -I.. -I..\third_party\include -Ic:\Python37-64\include -Wno-attributes -lxed -L..\third_party\lib -o microx_core.pyd

```

Comment:
The compilation should work through `setup.py` (and perhaps it does because there is nothing special in this `g++` command) when the python package is installed with `pip install <microx_repo_root_dir_or_url>` or `setup.py install`. That makes installation much easier and independent of python version and location.

Reply:
See https://stackoverflow.com/a/33618532/1806760, the users has to manually specify the compiler anyway so a `pip install whatever` will never work properly with MinGW

Reply:
Specifying the compiler is normal and done quite often in the unix world (but usually with the `CC` and `CXX` env vars). It is still worth using `pip install` or `setup.py install` because:

- It uses the correct python directory and include files. A machine can have many different python versions and many copies of a given version through python virtualenvs. Python developers often have at least one python virtualenv per project.
- It doesn't require the user to clone the `microx` package and mess with scripts: `pip install https://github.com/lifting-bits/microx` (by setting compiler with env vars or flags) installs `microx` to the correct python installation or virtualenv where the given `pip` command belongs.

Reply:
Yeah I'm aware this is common in the unix world, but it isn't common for Windows. In theory a motivated person and make this work from the command line, but I just thought to provide my local build script that I used for testing. I can remove it if you like.

Reply:
Installing a package and all of its dependencies with a single `pip` command applies to windows too.

Unfortunately native lib dependencies of python packages (in general, not only in case of XED) have to be installed manually but after installing them (and making them visible to the C/C++ compiler via system include+lib path or `CFLAGS` env var) a single `pip` command should be able to install huge chains/trees of packages without manually fiddling with the python package repos.

Reply:
I have deleted the build script from the repository.

Reply:
I tried to make it work, but mingw is no longer supported as of python 3.4+ (https://stackoverflow.com/questions/16737260/how-to-tell-distutils-to-use-gcc)

Reply:
C/C++ is the only programming language that I prefer using on windows because nothing really beats the Visual Studio + Visual Assist X plugin combo. In my opinion Visual Studio is by far the best choice on windows (both the IDE and the compiler), I always had a lot of trouble with alternatives mostly because of missing headers/definitions and the fact that a lot of windows compatible sources rely on MSVC features (even if unnecessarily).

Currently Visual Studio Community is free even for commercial purposes (for individuals and up to 5 people for smaller companies AFAIK) and unlike older versions, the current Community edition of the IDE allows the installation of addons/plugins (like VAssistX) so it's likely to be a significantly better option than the opensource alternatives in a lot of cases.

For the above reasons I'm wondering why are you using mingw... Especially because your code relies on very basic definitions that probably compiles with MSVC too.

Reply:
MSVC does not support ineline assembly.